### PR TITLE
update rust dependencies

### LINKED ISSUE #1933: pin tuftool to known good version
Currently we install `tuftool` via [cargo install tuftool](https://github.com/bottlerocket-os/bottlerocket/blob/877238719314e22faf13ba109a6552d5f0877c6b/Makefile.toml#L700), which means that every new release is immediately picked up. It also means that older commits might become unbuildable if we make a backwards incompatible change.

We should pin `tuftool` to a specific version so that our builds are more reliable and reproducible.

### PR DESCRIPTION
**Pull request by webern**
**Issue number:**

Closes #1933

**Description of changes:**

Update Rust dependencies they said. It will be easy I said.

**Testing done:**

Running some tests now

- [x] CI is happy with all builds
- [x] Ran aarch64 aws-ecs-1
  - [x] ran a task 
  - [x] logged in vis ssm enabled admin container and used sudo sheltie
  - [x] used apiclient to change the motd setting
- [x] x86 aws-k8s-1.21, Sonobuoy quick 
 
**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Review by jpculp — APPROVED** [review]
Built an aws-k8s-1.21 (x86_64) AMI on my side and it looks good!

**Review by etungsten — APPROVED** [review]

File: Makefile.toml [hunk 1]
```diff
@@ -704,6 +704,7 @@ cargo install \
   --jobs ${BUILDSYS_JOBS} \
   --root tools \
   --quiet \
+  --version 0.7.1 \
```
Comment (cbgbt) [inline Makefile.toml]: Should this be a variable in the Makefile? I worry we'll forget that this is buried in here.

**Review by zmrow — APPROVED** [review]
:juggling_person:

**Comment by webern** [comment]
^ make the tuftool version a Makefile variable.

**Review by etungsten — APPROVED** [review]

**Comment by webern** [comment]
^ undo variants/Cargo.lock change (???)

**Review by cbgbt — APPROVED** [review]

**Comment by jpculp** [comment]
I think you want to keep the `version = 3` in the `variants/Cargo.lock`. We already have it in tools and sources.

**Pull request merged**